src/onTTFB.ts

```diff
@@ -55,13 +60,13 @@ const whenReady = (callback: () => void) => {
  * includes time spent on DNS lookup, connection negotiation, network latency,
  * and server processing time.
  */
-export const onTTFB = (onReport: ReportCallback, opts?: ReportOpts) => {
+export const onTTFB = (onReport: TTFBReportCallback, opts?: ReportOpts) => {
   // Set defaults
   opts = opts || {};
 
   let metric = initMetric('TTFB');
   let report = bindReporter(
-    onReport,
+    onReport as ReportCallback,
```

Comment:
I think this is probably fine for now since this is internal code...

...but if we're going to be exporting metric-specific versions of all of these types, it seems like ideally we would be able to make the `bindReporter()` function understand those types as well. E.g. something similar to what we do here: https://github.com/GoogleChrome/web-vitals/blob/1d2c191e6763ea3ebeef05e9141847dbce554228/src/lib/observe.ts#L40-L44

Reply:
I couldn't figure out a way to do this as the metrics are not the same (particularly attribution has an extra `attribution` attribute). But open to pointers here.

Also, unlike in the `observe` example (where the specific type had to be passed on to the observer), we're treating all the metrics the same in `bindreporter`, so seemed easier to just assert this as a ReportCallback type and not treat them as the 12 individual types (6 metrics, and 6 metrics with attribution).

Reply:
Yeah, it's the mismatch between the specific callbacks for the `on*` methods (e.g. `TTFBReportCallback` for `onTTFB()`), and the fact that `bindReporter()` currently requires _its_ `onReport` to be able to handle _any_ of the metrics (which e.g. `TTFBReportCallback` purposefully doesn't) that you get the type error and the need for the type assertion.

I'm happy to take a stab at parameterizing `bindReporter` after this PR lands. Because `bindReporter` truly doesn't care about the handler internally, it should be relatively disruption-free type rearranging (and I believe should streamline some metric type stuff at the same time).

Reply:
PR in #359